include/kb_client.hpp

```diff
@@ -85,11 +100,20 @@ class Object {
      * std::bad_cast if the cast fails.
      */
     template<typename T>
-    T as() { return value_.as<T>(); }
+    T as() const { return value_.as<T>(); }
 
-    msgpack::object get() const { return value_; }
+    std::string dtype() const { return dtype_; }
 
-    std::string dtype() const { return msgpack_type_map.at(value_.type); }
+    std::size_t size() const { return size_; }
+
+    std::string subType() const {
+        if (!size_) return "";  // NIL
+        if (size_ == 1) return dtype_;
+        if (dtype_ == "MSGPACK_OBJECT_ARRAY") return msgpack_type_map.at(value_.via.array.ptr[0].type);
```

Comment:
is there a type if the array has no elements?

Reply:
No!  At least I do not know! This corner case is guarded by the first condition if (!size_) return ""